app/components/add-button.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
+import { Button } from "./ui/button";
+
+interface AddButtonProps {
+  initialCount?: number;
+  minCount?: number;
+  maxCount?: number;
+}
```

Comment:
To jest źle - to rodzic powinien mieć ten stan i przekazać do tego komponentu count i setState do propsa, który powinien się nazywać onCountChange. Rodzic przekazuje obecny stan a dziecko notyfikuje go (poprzez wywołanie funkcji onCountChange), że użytkownik kliknał w ten przycisk